File: 0011-container-with-most-water/brute-force/brute-force.cpp
using namespace std;
class Solution
{
    public:
        int maxArea(vector<int>& height)
        {
            int n = height.size();
            int max_area = 0;

            for (int i = 0; i < n; i++)
            {
                for (int j = 0; j < n; j++)
                {
                    int area = (j - i) * min(height[i], height[j]); // Area = width * height
                    max_area = max(area, max_area);
                }
            }

            return max_area;
        }
};

File: 0011-container-with-most-water/brute-force/brute-force.rs
use std::cmp;

impl Solution
{
    pub fn max_area(height: Vec<i32>) -> i32
    {
        let n: usize = height.len() as usize;
        let mut max_area: i32 = 0;

        for i in 0..n
        {
            for j in 0..n
            {
                let a: i32 = i as i32;
                let b: i32 = j as i32;

                let area: i32 = (b - a) * cmp::min(height[i], height[j]); // Area = width * height
                max_area = cmp::max(area, max_area);
            }
        }

        return max_area;
    }
}